evaluator/builtin_time.go

```diff
@@ -673,6 +675,414 @@ func builtinFromUnixTime(args []types.Datum, _ context.Context) (d types.Datum,
 	return builtinDateFormat([]types.Datum{d, args[1]}, nil)
 }
 
+func strToDate(t *time.Time, date string, format string) error {
+	if len(format) == 0 {
+		return nil
+	}
+
+	token, format1, err := getFormatToken(format)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if token == "" {
+		if date == "" {
+			return nil
+		}
+		return errors.New("not enough data in date")
```

Comment:
I test mysql behavior and find it never report error, just return NULL on fail.
I'll change to that.